src/sync/pull.ts

```diff
@@ -447,9 +458,9 @@ async function addChangedKeysForIndexes(
   for (const [newIndexName, newIndex] of newIndexes) {
     // new index name is not in the old indexes. All keys changed!
     await newIndex.withMap(read, async newMap => {
-      const changedKeys = await allKeys(newMap);
+      const changedKeys = await diffForAll(newMap);
       if (changedKeys.length > 0) {
-        changedKeysMap.set(newIndexName, await allKeys(newMap));
+        diffsMap.set(newIndexName, await diffForAll(newMap));
```

Comment:
We don't need to compute diffForAll again, we can just use the value computed above (currently named changedKeys, but should be named diffs).